pkg/auth/providers/saml/saml_provider.go

```diff
@@ -358,8 +358,13 @@ func (s *Provider) SearchPrincipals(searchKey, principalType string, token v3.To
 		principalType = "user"
 	}
 
+	name := s.userType + "://" + searchKey
+	if principalType == "group" {
+		name = s.groupType + "://" + searchKey
+	}
```

Comment:
nit: I'd probably just translate the principalType to either `s.userType` or `s.groupType` and combine in the literal definition below rather than duplicate the entire line twice.

So, something like:

```go
scheme := s.userType
if principalType == "group" {
  scheme = s.groupType
}

ObjectMeta:    metav1.ObjectMeta{Name: scheme + "://" + searchKey},
```